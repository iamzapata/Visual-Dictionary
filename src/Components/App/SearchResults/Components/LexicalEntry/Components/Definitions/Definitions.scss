.Senses {
  .Example {
    font-weight: 600;
  }
}
