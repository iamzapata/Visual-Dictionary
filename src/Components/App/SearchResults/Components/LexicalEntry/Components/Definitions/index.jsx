import React from "react"
import { shape, string, arrayOf } from "prop-types"
import "./Definitions.sass"

const Definitions = ({ entry }) => {
  const senses = entry.entries[0].senses.map(en => ({
    id: en.id,
    definitions: en.definitions,
    examples: en.examples,
    notes: en.notes
  }))

  return (
    <div className="Senses">
      <ul>
        {senses.map(sense => {
          return (
            <li key={sense.id} className="Sense">
              <span>
                {sense.notes && (
                  <i>[{sense.notes.map(n => n.text).join(",")}]</i>
                )}
              </span>
              <span>
                {sense.definitions &&
                  sense.definitions.map(def => (
                    <p className="Definition" key={def}>
                      {def}.
                    </p>
                  ))}
                <span>
                  {sense.examples &&
                    sense.examples.map(ex => (
                      <p className="Example" key={ex.text}>
                        ‘{ex.text}’
                      </p>
                    ))}
                </span>
              </span>
            </li>
          )
        })}
      </ul>
    </div>
  )
}

Definitions.propTypes = {
  entry: shape({
    id: string,
    definitions: arrayOf(shape),
    examples: arrayOf(shape),
    notes: arrayOf(shape)
  }).isRequired
}

export default Definitions
