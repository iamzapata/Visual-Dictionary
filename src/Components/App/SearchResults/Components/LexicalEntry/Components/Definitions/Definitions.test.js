import React from "react"
import { render, cleanup } from "react-testing-library"
import Definitions from "./index"

describe("<Definitinos />", () => {
  afterEach(cleanup)

  const entry = {
    entries: [{ senses: [] }]
  }

  it("Render self", () => {
    const { container } = render(<Definitions entry={entry} />)

    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
