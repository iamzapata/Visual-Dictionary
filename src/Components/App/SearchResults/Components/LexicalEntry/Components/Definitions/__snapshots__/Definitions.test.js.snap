// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Definitinos /> Render self 1`] = `
<div
  class="Senses"
>
  <ul />
</div>
`;
